[WIP] Establish Dashboard UI

angelocordon opened the pull request:
Define dashboard UI and components (#16) along with early stages of UX flow for connecting with other users.

TODO:
- [x] Define sidebar functionality to only display active sections
- [x] Render cards conditionally or message when no match cards are available
- [ ] UI Tests, Unit Tests (#41)
- [ ] Create a way to add a user card into a section (#42)
- [ ] ~Create functionality for `Request Connection` (needs discussion) (SEPARATE GITHUB ISSUE)~
- [ ] ~Create functionality for reporting a card (needs discussion) (SEPARATE GITHUB ISSUE)~
- [x] Connect Meteor / MongoDB data (needs discussion / help) 

![image](https://user-images.githubusercontent.com/3891869/54414145-663fe880-46b5-11e9-9e4c-03aa7a2950b9.png)

angelocordon committed:
Fix reference errors in routes

angelocordon committed:
Refactor Dashboard UI

angelocordon committed:
Refactor DashboardSidebar

Convert DashboardSidebar to a class component; will need to eventually bind
events and props from the Dashboard component.

Loop through an array of list items to render ListGroup.Item component.

angelocordon committed:
Refactor Dashboard component

Move Dashboard component to folder to refactor paths / routes

angelocordon committed:
Remove console.log in navbar jsx

angelocordon committed:
Add SCSS styles

angelocordon committed:
Refactor section to card component

angelocordon committed:
Refactor MatchCards into component

angelocordon committed:
Add MatchesSection

The MatchesSection component displays a user's current and past matches.

angelocordon committed:
Minor UI edits

angelocordon commented:
Currently at early stages of the Dashboard UI - wanted to see if I can get some early feedback to see if I'm going towards the right direction or if I'm missing anything so far. At some point, I'll have to figure out how to get Meteor or MongoDB data in here, and could use some discussions around that, along with other functionalities that some of the components need to do.

angelocordon committed:
Clean up Dashboard Index file

angelocordon committed:
Add Collapse functionality

angelocordon committed:
Set up target sections to be visible

Define target sections to be visible when clicking on a menu item in sidebar.
Clicking on Mentors should display sections for people looking for mentees.

Commit:
Created a DashboardContainer

Create DashboardContainer to load in entries into the dashboard context
Created a Profile component to load in DashboardContainer.

Commit:
Created a DashboardContainer and loaded in entries into the dashboard context. Also created a Profile component to load in DashboardContainer.

angelocordon commented on imports/ui/components/board/board.jsx:
```diff
@@ -6,8 +6,9 @@ import '/imports/ui/styles/_board.scss';
-    super(prop);
+  constructor(props) {
+    super(props);
+    console.log(props) //comes from BoardContainer
```
Let's try our best not to commit `console.log`s in ;) -- makes it harder to clean up when being pulled elsewhere

angelocordon committed:
Moved `sectionTargets` outside of Dashboard component class

Because these are static variables, it's best to move these out of the
`getTargetSections()` function so that they don't get recreated every time
the function is invoked.

Commit:
Merge branch 'user-dashboard-ui' of https://github.com/codebuddies/cb-connect into user-dashboard-ui

Commit:
Pass category ID number from state up into withTracker

Commit:
Pass down entries from withTracker to dashboard_cards_section

Commit:
Make sure to pass down 'all' entries if 'all' is selected

Commit:
Pass entries down into cards, and display default text if no entries found

Commit:
Fix section header names -- need to be the opposite of what the user selected

Commit:
Added _cards.scss style and added CardDeck

Commit:
Fix bug when 'all' is clicked on

Commit:
Show current unmathced cards created by user, and prevent button from appearing w/ a prop on that cards component

Commit:
Apply CSS fix suggestion from @angelocordon

nalbina commented on imports/ui/components/dashboard/dashboard.jsx:
```diff
@@ -0,0 +1,93 @@
+    const allSectionsVisible = this.state.visibleSections === 'all'
+    const sectionIsVisible = this.state.visibleSections === sectionKey
+    if (allSectionsVisible || sectionIsVisible) { return true }
+    return false
```
could be `return allSectionsVisible || sectionIsVisible`

Commit:
move timezone to the bottom of the card and change 'matched' to be a string

Commit:
comment out the console logs

Commit:
Remove limit on each category
Remove limit on category until pagination can be limited

Commit:
Rename 'Request Connection' to be 'Request Match'

Commit:
Rename 'past matches' to be 'Other Submissions'

Commit:
Deconstruct props in match_cards.js

Commit:
Use bootstrap helper class instead of defining .card-deck .card margin

Commit:
Use entry ID as key instead of counter index as key

Commit:
Remove feedback/career advice as a category

Commit:
Remove the unimportant console logs

Commit:
Fix issue with all entries displaying under each section when 'all' is clicked on
- Fix spelling of 'label'
- Use the imported categories.js when available
- Give 'Other' a category ID of 2
- Pass currentUserName down as a context variable

Commit:
Prevent users' own cards from being displayed when 'all' is clicked on in the sidebar, and remove  CSS styling

Commit:
Break 'entries' line to be more readable
